### DISCUSSION TITLE
Batch convert TIFFs to JPEG

### DISCUSSION THREAD

**Discussion by 2fl0wer** [ImageMagick/ImageMagick#4564] [Help] [answered]
Hi folks

I'm trying to use ImageMagick 7.1.0-16 Q16-HDRI x86_64 2021-11-20 to convert a collection of .tif files to .jpg but I'm having a small problem:

We start with the following TIFFs:

mike@Fennel:~/Pictures/00EarlyUndated/Film 02$ ls
02.tif  05.tif  07.tif  09.tif  11.tif   12.tif
03.tif  06.tif  08.tif  10.tif  12A.tif

For which I ten issue the following command:

mike@Fennel:~/Pictures/00EarlyUndated/Film 02$ magick *.tif *.jpg

I want this to produce JPEGs with the same filenames as the original TIFFs but obviously, with the '.jpg' extension, instead I get:

mike@Fennel:~/Pictures/00EarlyUndated/Film 02$ ls
 02.tif   07.tif    '*-10.jpg'   12.tif    '*-4.jpg'  '*-8.jpg'
 03.tif   08.tif     10.tif     '*-1.jpg'  '*-5.jpg'  '*-9.jpg'
 05.tif   09.tif     11.tif     '*-2.jpg'  '*-6.jpg'
 06.tif  '*-0.jpg'   12A.tif    '*-3.jpg'  '*-7.jpg'

What am I doing wrong please?

I tried using '??.tif' and '??.jpg' but this didn't improve matters.

I admit I'm no BASH expert, or Linux expert in general, but I thought this should work.

TIA for your help.

Mike H

**Comment by snibgo** [comment, marked as answer]
Your command reads all the tif files into memory, then tries to write them all out to a single file named "\*.jpg". As there can only be one file with that name, it appends a number to the filename and names them "\*-0.jpg" and "\*-1.jpg" and so on.

But that isn't what you want to do. And \* has a special meaning in forum software, which has messed with your file listing.

Instead, I suggest you use "magick mogrify", like this:
```
magick mogrify -format jpg *.tif
```
See the documentaton at https://imagemagick.org/script/mogrify.php

**Reply by 2fl0wer** [reply]
Thanks, that worked. The likes of mogrify looked too complex for what I was trying to do but not only did it work, I no longer got an error message for the TIFFs from Magick.
